internal/cli/atlas/accesslists/create.go

```diff
@@ -80,26 +83,85 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func IPAddress() (string, error) {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP, nil
+	}
+
+	return "", errors.New("not able to find your public IP address. Specify the public IP address for this command")
+}
+
+func (opts *CreateOpts) needsArg() bool {
+	// Unless currentIP flag is enabled and type is ip address, args are required.
+	return !(opts.entryType == ipAddress && opts.currentIP)
+}
+
+func (opts *CreateOpts) validateCurrentIPFlag(cmd *cobra.Command, args []string) func() error {
+	return func() error {
+		if !opts.needsArg() && len(args) > 0 {
+			return fmt.Errorf(
+				"please either provide [entry] or use %s to use current IP Address.\n\nUsage: %s",
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
+
+		if opts.needsArg() && len(args) == 0 {
+			return fmt.Errorf(
+				"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				1,
+				len(args),
+				cmd.UseLine(),
+			)
+		}
+
+		if opts.needsArg() && opts.currentIP {
```

Comment:
[q] should we simplify this? the current implementation is
```go
		if opts.entryType != ipAddress || opts.currentIP && opts.currentIP {
```

Reply:
from DeMorgan's Law, I think the current implementation is equivalent to:
```go
if (opts.entryType != ipAddress || !opts.currentIP) && opts.currentIP {
```

we can simplify indeed with
```go
if opts.entryType != ipAddress && opts.currentIP {
```

Thanks, it's more readable now 😃